CRM/Core/Resources.php

```diff
@@ -132,7 +132,9 @@ public function __construct($extMapper, $strings, $cacheCodeKey = NULL) {
     $this->strings = $strings;
     $this->cacheCodeKey = $cacheCodeKey;
     if ($cacheCodeKey !== NULL) {
-      $this->cacheCode = Civi::settings()->get($cacheCodeKey);
+      // AngularJS json partials are language-specific because they ship with the strings
+      // for the current language.
+      $this->cacheCode = Civi::settings()->get($cacheCodeKey) . CRM_Core_I18n::getLocale();
```

Comment:
Oh, I like the general idea of incorporating the locale into the browser-side cache-busting code. It seems like it could make several kinds of resources more amenable to multilingual.

Applying the trick in this spot means it applies once - at the start of the page-request. These values might rotate during the page-request (e.g. if the cache-code needs to be expired/reset, or if the locale changes).

Haven't tested, but I think this might be more responsive:

```diff
diff --git a/CRM/Core/Resources.php b/CRM/Core/Resources.php
index 0175befd74..4e93411b04 100644
--- a/CRM/Core/Resources.php
+++ b/CRM/Core/Resources.php
@@ -334,9 +334,11 @@ class CRM_Core_Resources implements CRM_Core_Resources_CollectionAdderInterface
 
   /**
    * @return string
+   *   A random-looking string that may be appended to resource-URLs.
+   *   This should be URL-safe (ie use alphanumerics and underscores).
    */
   public function getCacheCode() {
-    return $this->cacheCode;
+    return $this->cacheCode . CRM_Core_I18n::getLocale();
   }
 
   /**
@@ -563,7 +565,7 @@ class CRM_Core_Resources implements CRM_Core_Resources_CollectionAdderInterface
     $hasQuery = strpos($url, '?') !== FALSE;
     $operator = $hasQuery ? '&' : '?';
 
-    return $url . $operator . 'r=' . $this->cacheCode;
+    return $url . $operator . 'r=' . $this->getCacheCode();
   }
 
   /**
```